errors.go

```diff
@@ -27,6 +27,10 @@ var (
 	// progress because there is no change in location after Linesearcher step
 	// due to floating-point arithmetic.
 	ErrNoProgress = errors.New("linesearch: no change in location after Linesearcher step")
+
+	// ErrMaximumStep signifies that a Linesearcher reached a maximum allowed
+	// step that is not satisfactory.
+	ErrMaximumStep = errors.New("linesearch: reached maximum step")
```

Comment:
How about this is "ErrLinesearchBound". Some linesearchers have a minimum step, and this may as well cover both?